src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -246,5 +246,71 @@ void Add(INamedTypeSymbol symbol)
 
             return builder?.ToImmutable() ?? ImmutableArray<INamedTypeSymbol>.Empty;
         }
+
+        private static bool IsMethodNotImplemented(OperationBlockAnalysisContext context)
```

Comment:
Instead of cloning this code in 2 analyzers, can you make it an extension method on OperationBlockAnalysisContext (you can create a new extension file [here](https://github.com/dotnet/roslyn-analyzers/tree/master/src/Utilities/Extensions))?

Additionally, the current code for detection looks fragile as the IOperation tree shape is liable to change in future. Can you file a separate issue to think about a different strategy to implement it?

Reply:
Moved common code to extension and updated code accordingly. 
Regarding `IOperation` tree shape, please file it - as I don't know much about its details.